packages/main/src/RadioButtonGroup.ts

```diff
@@ -118,7 +118,11 @@ class RadioButtonGroup {
 
 		group.filter(radioBtn => !radioBtn.disabled).forEach((radioBtn, idx) => {
 			if (hasCheckedRadio) {
-				radioBtn._tabIndex = radioBtn.checked ? "0" : "-1";
+				if ((document.activeElement as RadioButton).readonly) {
```

Comment:
document.activeElement won't work correctly if the radio button is inside another shadow dom (composite component).

Also radioBtn._tabIndex = radioBtn.readonly ? "0" : "-1"; won't work if there are more than one readonly radio btn.

You can modify the code to:

```
group.filter(radioBtn => !radioBtn.disabled).forEach((radioBtn, idx) => {
			let activeElement : Element | RadioButton = getActiveElement()!;
			if (activeElement.classList.contains("ui5-radio-root")) {
				activeElement = activeElement.getRootNode() as Element;
				if (activeElement instanceof ShadowRoot) {
					activeElement = activeElement.host;
				}
			}

			if (hasCheckedRadio) {
				if (activeElement.hasAttribute("ui5-radio-button") && (activeElement as RadioButton).readonly) {
					radioBtn._tabIndex = activeElement === radioBtn && radioBtn.readonly ? "0" : "-1";
				} else {
					radioBtn._tabIndex = radioBtn.checked ? "0" : "-1";
				}
			} else {
				radioBtn._tabIndex = idx === 0 ? "0" : "-1";
			}
		});
```